### DISCUSSION TITLE
Does changing profiles actually alter *LIBL on server connection jobs?

### DISCUSSION THREAD

**Discussion by m-tyler** [codefori/vscode-ibmi#639] [General]
I have a couple of connection profiles with different "USER LIBRARY LIST"s.  I opened a work space .SQL tab so that I could run this statement, `select * from QSYS2.LIBRARY_LIST_INFO where TYPE in ('CURRENT','USER')`.  When I run this statement on each connection profiles, I only see the *LIBL of the base connection based on my user profile.   If I run this command through the SQL action (CTRL+R), `CL:DSPLIBL`, the resulting output shows the correct *LIBL according to the current connection setting seen in "USER LIBRARY LIST".  

I am going to assume the run of SQL statements is done through a different server job that must activate each time the statement is run where as the `CL:DSPLIBL` is run in a constantly connected server job.

**Comment by worksofliam** [comment]
The library list is stateless and when the user changes it, the extension does not change it on any server job.

Every ILE command is run in a new job (due to the nature of SSH), and we set it each time a command is run.

**Reply by m-tyler** [reply]
QSH utility listlist -d allows for a list of libraries, so this also means you can use a list of libraries to add. 

So I wonder how much effort it would be to save the in effect *LIBL, use liblist -d [USRLIBL], then use liblist -a [CONNECTION_LIBL] each time the connection profile is changed?   Would this even work?

**Reply by worksofliam** [reply]
I don't think it will work. It also wouldn't have any impact or change if it did work like that.

Perhaps I am not following.

**Reply by m-tyler** [reply]
My question is mostly due to ignorance of the tool design.   I actually did find several calls to the server that were doing a delete and re-add.  I am still getting used to the way it functions.    

As an RDi user, things look and act differently.  I have command actions that I use to set the *LIBL for my current filter work.   The Code for IBM i "Settings Profiles" seem like a better option for quickly switching environment settings.   

Plus, trying to get used to the internal SQL statement processing.   I quite often forget about the requirement of qualified objects in statements.  

So I guess my overall question has been answered.

**Reply by worksofliam** [reply]
@m-tyler I do wish I could re-write the way the SQL processor works. Long term we could actually write our own SQL processor that will apply the library list.. but that's a way away right now. Not enough time in the day to make awesome and useful improvements like that.

**Reply by m-tyler** [reply]
Oh, I completely understand.
